source/extensions/common/wasm/ext/envoy_proxy_wasm_api.cc

```diff
@@ -33,3 +33,7 @@ extern "C" PROXY_WASM_KEEPALIVE void envoy_on_resolve_dns(uint32_t context_id, u
                                                           uint32_t data_size) {
   getEnvoyContextBase(context_id)->onResolveDns(token, data_size);
 }
+
+extern "C" PROXY_WASM_KEEPALIVE void envoy_on_stat(uint32_t context_id, uint32_t data_size) {
+  getEnvoyContextBase(context_id)->onStat(data_size);
```

Comment:
If `onStat` callback is only supported on the `RootContext`, let's make it clear by calling something like `getEnvoyRootContext(..)` instead of `getEnvoyContextBase(..)`

Reply:
+1